### PR TITLE
Fix typing warnings

### PR DESCRIPTION
**Pull request by disrupted**
Fixed all warnings reported by pyright, which mypy doesn't detect

### CODE REVIEW

File: backend/streams_explorer/core/k8s_app.py [hunk 1]
```diff
@@ -132,7 +136,8 @@ def __get_common_configuration(self):
             elif name == self._get_env_name("EXTRA_OUTPUT_TOPICS"):
                 self.extra_output_topics = self.parse_extra_topics(env.value)
 
-            extractor_container.on_streaming_app_env_parsing(env, self.name)
+            if self.name:
+                extractor_container.on_streaming_app_env_parsing(env, self.name)
```
Comment (VictorKuenstler) [inline backend/streams_explorer/core/k8s_app.py:140]: Is the condition really needed? Name is not nullable, right?

**Review by VictorKuenstler — CHANGES_REQUESTED** [review]

File: backend/streams_explorer/core/services/dataflow_graph.py [hunk 2]
```diff
@@ -34,11 +34,12 @@ def add_streaming_app(self, app: K8sApp):
             node_type=NodeTypesEnum.STREAMING_APP,
             **app.attributes,
         )
-        self._add_topic(app.output_topic)
-        self._add_output_topic(app.name, app.output_topic)
+        if app.output_topic:
+            self._add_topic(app.output_topic)
+            self._add_output_topic(app.name, app.output_topic)
         if app.error_topic is not None:
```
Comment (yannick-roeder) [inline backend/streams_explorer/core/services/dataflow_graph.py:40]: Is `is not None` still necessary here?

File: backend/streams_explorer/py.typed [hunk 3]
```diff
@@ -0,0 +1 @@
+# Marker file for PEP 561. The streams-explorer package uses inline typing hints.
```
Comment (yannick-roeder) [inline backend/streams_explorer/py.typed:1]: Is this file needed in the repo?

File: backend/streams_explorer/core/k8s_app.py (hunk 1 above)
Comment (disrupted) [reply backend/streams_explorer/core/k8s_app.py]: Absolutely, this is now obsolete.

File: backend/streams_explorer/py.typed (hunk 3 above)
Comment (disrupted) [reply backend/streams_explorer/py.typed:1]: since we're publishing it as a library to PyPI I think it's good to follow the PEP guidelines

File: backend/streams_explorer/core/k8s_app.py [hunk 4]
```diff
@@ -15,26 +15,27 @@
...
+        self.name: str
+        self.metadata: Optional[V1ObjectMeta] = None
+        self.input_topics: List[str] = []
+        self.output_topic: Optional[str] = None
+        self.error_topic: Optional[str] = None
+        self.extra_input_topics: List[str] = []
+        self.extra_output_topics: List[str] = []
         self.attributes: Dict[str, str] = {}
 
     def to_dict(self) -> dict:
```
Comment (VictorKuenstler) [inline backend/streams_explorer/core/k8s_app.py:27]: Decorate it with `@abstractmethod`?

File: backend/streams_explorer/core/k8s_app.py [hunk 5]
```diff
@@ -15,26 +15,27 @@
...
-        pass
+        return {}
 
-    def get_name(self) -> Optional[str]:
-        pass
+    def get_name(self) -> str:
```
Comment (VictorKuenstler) [inline backend/streams_explorer/core/k8s_app.py:30]: Same as above?

**Review by VictorKuenstler — CHANGES_REQUESTED** [review]

File: backend/streams_explorer/core/k8s_app.py (hunk 4 above)
Comment (disrupted) [reply backend/streams_explorer/core/k8s_app.py:27]: Without deriving from ABC, `@abstractmethod` is not permitted

**Review by VictorKuenstler — APPROVED** [review]
